internal/pkg/testutils/testutils.go

```diff
@@ -44,3 +46,17 @@ func NewErrorLogger(t *testing.T) *logger.Logger {
 	require.NoError(t, err)
 	return log
 }
+
+// SkipIfFIPSOnly will mark the passed test as skipped if GODEBUG=fips140=only is detected.
+// If GODBUG=fips140=on, go may call non-compliant algorithms and the test does not need to be skipped.
+func SkipIfFIPSOnly(t *testing.T, msg string) {
+	// NOTE: This only checks env var; at the time of writing fips140 can only be set via env
+	// other GODEBUG settings can be set via embedded comments or in go.mod, we may need to account for this in the future.
+	ss := strings.Split(os.Getenv("GODEBUG"), ",")
+	for _, s := range ss {
+		if s == "fips140=only" {
```

Comment:
This should probably be a `string.Contains` check instead for robustness. Other options could be set as well.